compiler/src/iree/compiler/Dialect/Flow/Transforms/DumpDispatchGraph.cpp

```diff
@@ -0,0 +1,422 @@
+// Copyright 2022 The IREE Authors
+//
+// Licensed under the Apache License v2.0 with LLVM Exceptions.
+// See https://llvm.org/LICENSE.txt for license information.
+// SPDX-License-Identifier: Apache-2.0 WITH LLVM-exception
+
+//===------------------- DumpDispatchGraph.cpp ----------------------------===//
+//
+// Generate a graphviz graph for dispatches
+//
+//===----------------------------------------------------------------------===//
+
+#include <utility>
+
+#include "PassDetail.h"
+#include "iree/compiler/Dialect/Flow/Transforms/PassDetail.h"
+#include "iree/compiler/Dialect/Flow/Transforms/Passes.h"
+#include "llvm/Support/Format.h"
+#include "llvm/Support/GraphWriter.h"
+#include "mlir/Dialect/Arithmetic/IR/Arithmetic.h"
+#include "mlir/Dialect/Func/IR/FuncOps.h"
+#include "mlir/IR/AsmState.h"
+#include "mlir/IR/Block.h"
+#include "mlir/IR/BuiltinTypes.h"
+#include "mlir/IR/Operation.h"
+#include "mlir/Support/IndentedOstream.h"
+
+namespace mlir {
+namespace iree_compiler {
+namespace IREE {
+namespace Flow {
+
+static const StringRef kLineStyleControlFlow = "dashed";
+static const StringRef kLineStyleDataFlow = "solid";
+static const StringRef kShapeNode = "box";
+static const StringRef kShapeBox = "box";
+static const StringRef kShapeTab = "tab";
+static const StringRef kShapeNone = "plain";
+static const StringRef kShapeEllipse = "ellipse";
+
+static StringRef getShape(Operation *op) {
```

Comment:
Actually I dont like the node shapes. LLVM uses records for the shape. It is much easier to read that way (https://graphviz.org/doc/info/shapes.html#record) .

Reply:
To add to this, previously when I used graphviz (I spent a lot of time to get this readable), the format I generally used was 
- top row of the record for operands
- bottom row of the record for results

Then you can draw edges from result of one record to the operands of the other record. Accurately captures the dependence pattern. You can add more rows to the record of a node to describe the node, like op name etc.

This is very similar to what the visualization of selectionDAG from LLVM does (which is what I reverse engineered0.

Reply:
Thanks for the suggestion. This is a matter of style and it might need an option to represent a node as a record. One concern is more information in the node might make the dot rendering slow. I will handle the suggestion in a separate issue. Thanks for the review!